packages/commonwealth/server/workers/commonwealthConsumer/commonwealthConsumer.ts

```diff
@@ -74,6 +74,16 @@ export async function setupCommonwealthConsumer(): Promise<void> {
     BrokerSubscriptions.ContestWorkerPolicy,
     ContestWorker(),
     buildRetryStrategy(undefined, 20_000),
+    {
+      beforeHandleEvent: (topic, event, context) => {
+        context.start = Date.now();
+      },
+      afterHandleEvent: (topic, event, context) => {
+        const duration = Date.now() - context.start;
+        const handler = `${topic}.${event.name}`;
+        stats().histogram(`cw.handlerExecutionTime`, duration, { handler });
```

Comment:
what happens if we configure multiple event handlers with hooks? ... can we filter the stats by event handler name?

Reply:
Yep that's how it works– `handler` will be `"ContestWorkerPolicy.ThreadCreated"` or `"ContestWorkerPolicy.ThreadUpvoted"`. We can filter by handler in DD.

Currently only applies to contest worker policy handlers but we can apply to other handler by reusing the hooks object for other subscriptions.